### ISSUE TITLE
Receiving a new union type not included in the schema breaks parsing

### ISSUE THREAD

**Issue by mohamed-khaled-hsn** [awslabs/aws-mobile-appsync-sdk-android#313]
I'm not sure this is a bug or it is related to using an old version of Apollo in the SDK

Say that we have this defined:

schema:
```GraphQL
union Attachment = Event | Media
```

```GraphQL
fragment Attachment on Attachment {
    __typename
    ... on Event {
        __typename
    }
    ... on Media {
        __typename
        url
    }
}
```

Receiving a new union type in the response ex:
```
     "attachment": {
      "__typename": "BroadcastAttachment"
    }
```
Note `attachment` is specified as nullable i.e. `Attachment` not `Attachment!`

Causes `com.apollographql.apollo.exception.ApolloParseException: Failed to parse http response`
Caused by: java.lang.NullPointerException: attachment == null
and breaks parsing of the **whole** query so no result is returned

In the generated code I noticed this:

```java
      public static final class Mapper implements FragmentResponseFieldMapper<Fragments> {
        final fragment.Attachment.Mapper attachmentFieldMapper = new fragment.Attachment.Mapper();

        @Override
        public @Nonnull Fragments map(ResponseReader reader, @Nonnull String conditionalType) {
          fragment.Attachment attachment = null;
          if (fragment.Attachment.POSSIBLE_TYPES.contains(conditionalType)) {
            attachment = attachmentFieldMapper.map(reader);
          }
          return new Fragments(Utils.checkNotNull(attachment, "attachment == null"));
        }
      }

```
Basically what is expected is for new types that weren't specified in the fragment **and schema** they would be ignored
 
Is there any way apps that use the old schema can still work when the response contains a new type? 

Appsync version: `2.11.0`

The same setup is working and parsing correctly using the latest version of apollo `2.3.1` in a demo project

**Comment by jamesonwilliams** [comment]
> I'm not sure this is a bug or it is related to using an old version of Apollo in the SDK

Most likely. https://github.com/awslabs/aws-mobile-appsync-sdk-android/issues/303#issuecomment-668362275

**Comment by mohamed-khaled-hsn** [comment]
For those having the same issue, I noticed it is only happening with fragments that use union types, I used the contents of the fragment directly in the queries and it worked fine but it caused a lot of duplication.
